### ISSUE TITLE
Polaris fails to parse/ignore Kustomization yaml JSONPatch files

### ISSUE THREAD

**Issue by HariSekhon** [FairwindsOps/polaris#667]
### Installation Process
Docker image

### Polaris Versions
1.2, 2.0, 3.2.1

### Expected Behavior

It should have ignored the json patch format yaml files when it realized they weren't in the expected yaml format, or there should be an option to ignore files that match a given glob or regex.

### Actual Behavior

Polaris errored out and there doesn't appear to be any CLI way to get it to ignore these jsonpatch yaml files which are valid in Kustomize.

### Steps to Reproduce

```
docker run -ti \
  -v "$PWD:/pwd" \
  -v ~/.kube/config:/opt/app/config:ro \
  quay.io/fairwinds/polaris:1.2 polaris audit \
    --kubeconfig /opt/app/config \
    --audit-path /pwd 
ERRO[0000] Invalid YAML: - op: replace
  path: /spec/syncPolicy
  value: {} 
ERRO[0000] Error parsing YAML: (error unmarshaling JSON: while decoding JSON: json: cannot unmarshal array into Go value of type kube.k8sResource) 
ERRO[0000] Error fetching Kubernetes resources error unmarshaling JSON: while decoding JSON: json: cannot unmarshal array into Go value of type kube.k8sResource 

```

**Comment by sudermanjr** [comment]
Thanks for the detailed report. One question, does this actually stop parsing and fail out, or does it keep going and provide output?

**Comment by HariSekhon** [comment]
It stops dead.

This is why it needs at least an `--ignore` glob/regex switch to work around the problem for now.

**Comment by HariSekhon** [comment]
Btw you can find sample JSONPatch format yamls in my public kubernetes github repo here:

https://github.com/harisekhon/kubernetes-configs

In this case I believe it's failing on the `argocd-disable-autosync.jsonpatch.yaml` but this also happens in my private production directories for other patches like replica overrides between environments too and I know these patches work in Kustomize because I use them in production in multiple environments.

**Comment by intrand** [comment]
@HariSekhon I was able to reproduce your issue using the repo you provided. Thank you very much for that! I'm working on a fix. Will let you know when this is sorted out.

**Comment by intrand** [comment]
@HariSekhon You may track the progress of this fix with the PR linked above (#678). Thanks again for the report. I'll leave the issue open to give you time to test the fix once it's merged and released.
